### PR TITLE
Throw exception and give error message when creating schema with spaces in column name

### PR DESCRIPTION
**Pull request by mqliang**
## Description

Throw exception and give error message when creating schema with spaces in column name

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

**Comment by mqliang** [comment]
cc @fx19880617

File: pinot-core/src/test/java/org/apache/pinot/core/util/SchemaUtilsTest.java [hunk 1]
```diff
@@ -313,6 +315,24 @@ public void testDateTimeFieldSpec()
     SchemaUtils.validate(pinotSchema);
   }
 
+  @Test
+  public void testColumnNameValidation()
```
Comment (siddharthteotia) [inline pinot-core/src/test/java/org/apache/pinot/core/util/SchemaUtilsTest.java:319]: Let's add test for trailing and leading whitespace as well

File: pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java [hunk 2]
```diff
@@ -79,6 +81,8 @@ public static void validate(Schema schema) {
     for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
       if (!fieldSpec.isVirtualColumn()) {
         String column = fieldSpec.getName();
+        Preconditions.checkState(!StringUtils.containsWhitespace(column),
+            "The column name \"%s\" should not" + " contain blank space.", column);
```
Comment (siddharthteotia) [inline pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java:85]: (nit) Can be "should not contain blank space" instead of "should not" + " contain ...."

File: pinot-core/src/test/java/org/apache/pinot/core/util/SchemaUtilsTest.java (hunk 1 above)
Comment (siddharthteotia) [reply pinot-core/src/test/java/org/apache/pinot/core/util/SchemaUtilsTest.java:319]: Also please add a comment stating that currently the test is only for checking whitespaces. Later on if we add more validation on column names, we can corresponding tests here.

**Review by siddharthteotia — APPROVED** [review]
